bench/bench_support.lua

```diff
@@ -5,6 +5,16 @@ bench.runs = 20
 bench.extraRuns = 4
 
 function bench.runCode(f, description)
+    -- Under Callgrind, run the test only once and measure just the execution cost
+    if callgrind and callgrind("running") then
+        if collectgarbage then collectgarbage() end
+
+        callgrind("zero")
+        f() -- unfortunately we can't easily separate setup cost from runtime cost in f unless it calls callgrind()
```

Comment:
Setup code inside a benchmark can be placed outside the specific function that is being benchmarked.

Reply:
Yeah. We still have some benchmarks that don't do that and instead do internal time measurement and return it. Ideally we should probably just adjust those.